R/calendly.R

```diff
@@ -1,37 +1,137 @@
 # Calendly data extraction handlers
 
-#' Get Calendly API user
+#' Handle Calendly GET requests
 #' @description This is a function to get the Calendly API user info
-#' @param api_key You can provide the API key directly or this function will attempt to grab an API key that was stored using the `authorize("calendly")` function
+#' @param token You can provide the API key directly or this function will attempt to grab an API key that was stored using the `authorize("calendly")` function
+#' @param user The user param for Calendly. Usually looks like "https://api.calendly.com/users/c208a750-9214-4c62-9ee6-a1a9507c7b43"
+#' @param count For paginated GETs, you can specify how many things you'd like returned
+#' @param page_token For a paginated GET, what page are we on?
 #' @return Calendly REST API response as a list
 #' @importFrom utils menu installed.packages
 #' @importFrom httr oauth_app oauth_endpoints oauth2.0_token
 #' @export
 #' @examples \dontrun{
 #'
 #' authorize("calendly")
-#' get_calendly_user()
+#' token <- get_token(app_name = "calendly")
+#'
+#' result_list <- calendly_get(
+#'   url = "https://api.calendly.com/users/me",
+#'   token = token
+#' )
 #' }
-get_calendly_user <- function(api_key) {
-  # Get auth token
-  token <- get_token(app_name = "calendly")
-
-  # Declare URL
-  url <- "https://api.calendly.com/users/me"
-
-  # Github api get
+#'
+# Get Calendly stuffs
+calendly_get <- function(url, token, user = NULL, count = NULL, page_token = NULL) {
   result <- httr::GET(
     url,
+    query = list(
+      user = user,
+      count = count,
+      page = page_token
+    ),
     httr::add_headers(Authorization = paste0("Bearer ", token)),
     httr::accept_json()
   )
 
+  # Stop if error
   if (httr::status_code(result) != 200) {
     httr::stop_for_status(result)
   }
 
-  # Process and return results
   result_content <- httr::content(result, "text")
-  result_list <- jsonlite::fromJSON(result_content)
+  return(jsonlite::fromJSON(result_content))
+}
+
+#' Get Calendly API user
+#' @description This is a function to get the Calendly API user info
+#' @param api_key You can provide the API key directly or this function will attempt to grab an API key that was stored using the `authorize("calendly")` function
```

Comment:
Confused with the argument `api_key` as we are technically grabbing the personal access token. What about using `token` as an argument instead?

```suggestion
#' @param token You can provide the token directly or this function will attempt to grab the token that was stored using the `authorize("calendly")` function
```

Reply:
I have been inconsistent in this. At some times it's an api key and sometimes it's a token. I've been realizing this as I've been writing but I've not yet fixed it. Because you've noticed I'll make sure to fix it now lol

Reply:
Made it more consistent that we always refer to it as a "token" instead of an "api_key"